logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -15,6 +15,21 @@ def close
   end
 end
 
+class DummyInputGenerator < LogStash::Inputs::Base
```

Comment:
Well, lets keep the DummyGenerator, since we don't need a dependency for that and you were proposing to move away. This also in line with the actual dummy plugin in the current `pipeline_spec`